src/PublicApiAnalyzers/Core/Analyzers/DeclarePublicApiAnalyzer.cs

```diff
@@ -263,6 +298,16 @@ private static bool ValidateApiFiles(ApiData shippedData, ApiData unshippedData,
                 errors.Add(Diagnostic.Create(PublicApiFilesInvalid, Location.None, InvalidReasonShippedCantHaveRemoved));
             }
 
+            if (shippedData.NullableLines.Any(nl => nl.Rank != 0))
```

Comment:
Term `NullableLines` confused me into thinking these are lines which have nullable annotations. Given that you are using this only to determine if nullability annotations are enabled or not, can we instead just store a single int, say `NullabilityRank`, instead of all the lines enabling nullable? This should also allow you to report the `InvalidReasonMisplacedNullableEnable` diagnostics below.